Sources/ServiceLifecycle/ServiceGroupConfiguration.swift

```diff
@@ -41,17 +46,139 @@ public struct ServiceGroupConfiguration: Hashable, Sendable {
         public init() {}
     }
 
+    public struct ServiceConfiguration: Sendable {
+        public struct TerminationBehavior: Sendable, CustomStringConvertible {
+            internal enum _TerminationBehavior {
+                case cancelGroup
+                case gracefullyShutdownGroup
+                case ignore
+            }
+
+            internal let behavior: _TerminationBehavior
+
+            public static let cancelGroup = Self(behavior: .cancelGroup)
+            public static let gracefullyShutdownGroup = Self(behavior: .gracefullyShutdownGroup)
+            public static let ignore = Self(behavior: .ignore)
+
+            public var description: String {
+                switch self.behavior {
+                case .cancelGroup:
+                    return "cancelGroup"
+                case .gracefullyShutdownGroup:
+                    return "gracefullyShutdownGroup"
+                case .ignore:
+                    return "ignore"
+                }
+            }
+        }
+
+        /// The service.
+        public var service: any Service
+        /// The behavior when the service returns from its `run()` method.
+        public var successfulTerminationBehavior: TerminationBehavior
+        /// The behavior when the service throws from its `run()` method.
+        public var failureTerminationBehavior: TerminationBehavior
```

Comment:
mild nit: successful is an adjective, failure is a noun; can we have `success` instead of successful?